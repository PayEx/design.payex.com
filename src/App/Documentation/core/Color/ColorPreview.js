import React from "react";
import PropTypes from "prop-types";

const ColorPreview = ({ name, hex }) => {
    const style = { backgroundColor: hex };

    return (
        <div className="color-preview">
            <div className={`color-block${name === "white" ? " border" : ""}`} style={style}/>
            <p>
                {name}
                <small><br/>HEX: {hex}</small>
            </p>
        </div>
    );
};

ColorPreview.propTypes = {
    name: PropTypes.string.isRequired,
    hex: PropTypes.string.isRequired
};

export default ColorPreview;
