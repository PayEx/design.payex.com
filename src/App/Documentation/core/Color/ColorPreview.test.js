import React from "react";
import { shallow } from "enzyme";

import ColorPreview from "./ColorPreview";

describe("Core: Color", () => {
    it("is defined", () => {
        expect(ColorPreview).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<ColorPreview name="test" hex="#test" color="black" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("<p>test<small><br/>HEX: #test</small></p>");
        expect(wrapper.html()).toContain("background-color:#test");
    });
});
