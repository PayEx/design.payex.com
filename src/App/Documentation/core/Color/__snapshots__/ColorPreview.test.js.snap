// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: Color renders 1`] = `
<div
  className="color-preview"
>
  <div
    className="color-block"
    style={
      Object {
        "backgroundColor": "#test",
      }
    }
  />
  <p>
    test
    <small>
      <br />
      HEX: 
      #test
    </small>
  </p>
</div>
`;
